Ronin.rte/Devices/Tools/Shovel/Shovel.lua

```diff
@@ -79,13 +80,36 @@ function Update(self)
 
 			self.Magazine.RoundCount = resource > 0 and resource or -1;
 		end
-		self.RateOfFire = self.minimumRoF + (self.minimumRoF) * (actor.Health/actor.MaxHealth);
+		self.RateOfFire = self.minimumRoF + (self.minimumRoF) * (parent.Health/parent.MaxHealth);
 	else
 		self.Scale = 1;
 		if self.Magazine then
 			self.Magazine.Scale = 0;
 		end
+		if self.lastVel.Magnitude > 25 then
+			if self.HitWhatMOID ~= rte.NoMOID then
+				local mo = MovableMan:GetMOFromID(self.HitWhatMOID);
+				if mo then
+					local particleCount = 3;
+					local spread = self.AngularVel * TimerMan.DeltaTimeSecs * 0.5;
+					for i = 0, particleCount - 1 do
+						local damagePar = CreateMOPixel("Smack Particle");
```

Comment:
Couple things:
1. It might be advisable to create your particle in Create and store it to a self variable, then Clone it when you wanna use it. That way there's less presetman lookups and stuff.
2. In general when using CreateWhatever you probably wanna specify the datamodule. If you don't, the game looks through every preset in every datamodule until it finds a preset with the specified name, instead of just looking through the one you tell it to. If this thing exists in Base.rte I think there's effectively no difference, but otherwise your particle spawning is slower than it should be.

Reply:
1. Is this desirable for all similar accounts? In that case it would mean a lot of data refactoring, not to mention it would just feel more confusing to have those definitions imo, but if it's gainz then I'm not complaining.

2. yeah yeah

Reply:
Hmmm, so if you do specify a datamodule it's better, but still inherently gonna be slower than using clone directly.
Basically when you create something you tell PresetMan to do this
```
if (no datamodule specified) {
	for each datamodule, while there's no entity found {
		search for entity
	}
} else {
	search datamodule for entity
}
```
And datamodule search for entity does this
```
for each entity in datamodule of the appropriate class (i.e. AHuman, MOPixel, etc.)
	if entity.PresetName == specifiedPresetName && entity.ClassName == correctClassName {
		return entity
	}
}
return nothing
```

So if you have 500 particles to check against in your datamodule, it'll have to do a bit of work. A string check is super quick, so I don't think it's hugely worth changing this, unless you're making a lot of these things every frame, cause if you're making 100 and each one requires 500 * 2 string checks + other overhead, it might start to matter. Obviously you wouldn't be making 100 things every frame, but you might be making 10 or 20 for some things, which might be enough to make it worthwhile.